lib/flowutils.flow

```diff
@@ -50,64 +46,33 @@ eitherMapAnyType(x : flow, fn : (?) -> ??, alternative : ??) -> ?? {
 }
 
 buildKeyValueForPhp(key : string, value : ?) -> [KeyValue] {
-	if (isSameStructType(value, RequestIdsArray([])) || isSameStructType(value, RequestIdsRanges([]))) {
-		switch (flow(value)) {
-			RequestIdsArray(ids): [KeyValue(key, toStringNoExtraQuotes(ids))];
-			RequestIdsRanges(ranges): buildRangesKeyValueForPhp(key, ranges);
-		}
-	} else {
-		eitherMapAnyType(value, \v -> [KeyValue(key, toStringNoExtraQuotes(v))], []);
-	}
-}
-
-buildRangesKeyValueForPhp(key : string, ranges : [Pair<int, int>]) -> [KeyValue] {
-	values = map(ranges, \range -> {
-		if (range.first == range.second) flow(range.first) else flow([range.first, range.second])
-	});
-	[KeyValue(key, toStringNoExtraQuotes(values))];
+	eitherMapAnyType(value, \v -> [KeyValue(key, toStringNoExtraQuotes(v))], []);
 }
 
-idsArrayToRanges(ids : [int]) -> RequestIds {
-	if (ids != [] && isUrlParameterTrue("new")) {
-		sorted = sort(ids);
-		if (foldi(sorted, false, \i, acc, id -> acc || i > 0 && id - 1 == sorted[i - 1])) {
-			last = length(sorted) - 1;
-			list = foldi(sorted, makeList(), \i : int, acc : List<Pair<int, int>>, id : int -> {
-				if (i == 0 || (id - 1 != sorted[i - 1] && id != sorted[i - 1])) {
-					Cons(Pair(id, id), acc);
-				} else if (i < last && (id + 1 == sorted[i + 1] || id == sorted[i + 1])) {
+encodeRequestIdsForPhp(ids : [int]) -> string {
+	len = length(ids);
+	if (len > 1 && (isUrlParameterTrue("new") || isUrlParameterTrue("delta_encoding"))) {
+		prefix = "DE:";
+		lastIndex = len - 1;
+		"\"" + foldi(ids, Triple("", 0, 0), \i, acc, value -> {
+			if (i == 0) {
+				Triple(prefix, value, 1);
+			} else {
```

Comment:
You could avoid this `if` rewriting `fold`: `foldi(tail[ids], Triple(prefix, ids[0], 1), ....`